include/TFEL/Material/Eshelby.hxx

```diff
@@ -0,0 +1,108 @@
+/*!
+ * \file   include/TFEL/Material/Eshelby.hxx
+ * \author Antoine Martin
+ * \date   15 October 2024
+ * \brief  This file declares the Eshelby tensor for an ellipsoidal inclusion embedded in an isotropic matrix.
+ * \copyright Copyright (C) 2006-2018 CEA/DEN, EDF R&D. All rights
+ * reserved.
+ * This project is publicly released under either the GNU GPL Licence
+ * or the CECILL-A licence. A copy of thoses licences are delivered
+ * with the sources of TFEL. CEA or EDF may also distribute this
+ * project under specific licensing conditions.
+ */
+
+#ifndef LIB_TFEL_MATERIAL_ESHELBY_HXX
+#define LIB_TFEL_MATERIAL_ESHELBY_HXX
+
+#include "TFEL/Math/st2tost2.hxx"
+#include "TFEL/Material/StiffnessTensor.hxx"
+
+namespace tfel::material
+{
+  /*!
+   * This function builds the Eshelby tensor for an ellipsoid whose semi-axis lengths are a,b,c
+   * embedded in an isotropic matrix (young,nu). 
+   * It returns an object of type st2tost2, which is the fourth-order Eshelby tensor, in a basis which is adapted to the ellipsoid :
+   * see Eshelby.ixx for the details. The basis IS NOT RELATED to the given order of the values a,b,c
+   * These expressions can be found in Torquato 2002 Random Heterogeneous Materials for the axisymetric ellipsoid
+   * and in Eshelby 1957 for other cases.
+   * \tparam    real: underlying type
+   * param 1 : nu
+   * param 2 : semi-axis of ellipsoid in direction 1
+   * param 3 :semi-axis of ellipsoid in direction 2
+   * param 4 : semi-axis of ellipsoid in direction 3
+   */
+     template <typename real>
+  TFEL_HOST_DEVICE const tfel::math::st2tost2<3u,real> EshelbyTensor(
```

Comment:
`const` is not meaningful here.

```suggestion
     template <typename real>
  TFEL_HOST_DEVICE tfel::math::st2tost2<3u,real> EshelbyTensor(
```

`NumericType` could be better than `real`

Reply:
I suppressed const but I do not understand why it is useless, and also why NumericType is better